### PR TITLE
Update RegExp tests for the semantics change

### PR DESCRIPTION
**Pull request by littledan**
Spec change at https://github.com/tc39/ecma262/pull/798

Tested on V8's implementation of the new semantics.

cc @schuay @bterlson @anba @claudepache

### CODE REVIEW

**Comment by schuay** [comment]
Thanks for the quick turnaround, looks good to me.

**Comment by littledan** [comment]
I couldn't stand to see you checking in failed test262 status file lines, @schuay !

File: test/built-ins/RegExp/prototype/exec/failure-lastindex-access.js [hunk 1]
```diff
@@ -18,16 +19,18 @@ info: >
...
 
 var r = /a/;
-r.lastIndex = thrower;
+r.lastIndex = counter;
 
 var result = r.exec('nbc');
 assert.sameValue(result, null);
-assert.sameValue(r.lastIndex, thrower);
-
+assert.sameValue(r.lastIndex, counter);
```
Comment (leobalter) [inline test/built-ins/RegExp/prototype/exec/failure-lastindex-access.js]: nice approach, I would try some get/set accessor operation in there but this works and it's simple.

**Comment by leobalter** [comment]
LGTM

File: test/built-ins/RegExp/prototype/exec/failure-lastindex-access.js (hunk 1 above)
Comment (littledan) [inline test/built-ins/RegExp/prototype/exec/failure-lastindex-access.js]: That was my first idea, but I had forgotten that `lastIndex` is not configurable, so you can't make a getter or setter.

**Pull request merged**